File: test/com/jetbrains/android/dsl/tests/functional/PropertyTest.java
package com.jetbrains.android.dsl.tests.functional;

import com.jetbrains.android.dsl.BaseGeneratorProps;
import com.jetbrains.android.dsl.Subsystem;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.io.File;

public class PropertyTest extends BaseFunctionalTest {
    private final String testDataFile = "com/jetbrains/android/dsl/tests/functional/PropertyTest.kt";

    @Override
    @BeforeMethod
    public void setUp() throws Exception {
        super.setUp();
    }

    protected void initSettings(BaseGeneratorProps settings) {
        settings.setGenerateProperties(true);
        settings.setGeneratePropertySetters(true);
    }

    @Test
    public void testProperties() throws Exception {
        runFunctionalTest(testDataFile, Subsystem.PROPERTIES);
    }
}

File: test/org/jetbrains/android/dsl/functional/PropertyTest.kt
package org.jetbrains.android.dsl.functional
import org.jetbrains.android.dsl.*
import org.testng.annotations.Test
public class PropertyTest : AbstractFunctionalTest() {
    private val testDataFile = "PropertyTest.kt"
    override fun initSettings(settings: BaseGeneratorConfiguration) {
        settings.files.add(KoanFile.PROPERTIES)
    }
    [Test] public fun testPropertyTestFor21() {
        runFunctionalTest(testDataFile, KoanFile.PROPERTIES, "21")
    }
    [Test] public fun testPropertyTestFor21s() {
        runFunctionalTest(testDataFile, KoanFile.PROPERTIES, "21s")
    }
}
